### PR TITLE
Drop gradle5 support

### PR DESCRIPTION
**Pull request by maczikasz**
This PR will remove the code that was created for gradle 5 compatibilty support, it also updates the wrapper to the latest and the stutter matrix to remove 5.0 and add the latest gradle versions

There is currently an issue with running with Gradle 8.2 milestone-1 hence the compat tests were maximized at 8.1 (inclusive)

### CODE REVIEW

File: src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishExtension.kt [hunk 1]
```diff
@@ -72,6 +74,14 @@ open class NexusPublishExtension(project: Project) {
...
+    val publicationType: Property<PublicationType> = project.objects.property<PublicationType>().convention(PublicationType.MAVEN)
+
+    val ivyPatternLayout: Property<Action<IvyPatternRepositoryLayout>> = project.objects.property<Action<IvyPatternRepositoryLayout>>()
+    fun ivyPatternLayout(action: Action<IvyPatternRepositoryLayout>) {
+        ivyPatternLayout.set(action)
+    }
+
     fun repositories(action: Action<in NexusRepositoryContainer>) = action.execute(repositories)
 
     enum class PublicationType(internal val gradleType: KClass<out Publication>, internal val publishTaskType: KClass<out DefaultTask>) {
```
Comment (marcphilipp) [inline src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishExtension.kt]: Shouldn't this be moved to `NexusRepository`?

File: src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishExtension.kt [hunk 2]
```diff
@@ -72,6 +74,14 @@ open class NexusPublishExtension(project: Project) {
...
     )
 
+
+    val publicationType: Property<PublicationType> = project.objects.property<PublicationType>().convention(PublicationType.MAVEN)
+
+    val ivyPatternLayout: Property<Action<IvyPatternRepositoryLayout>> = project.objects.property<Action<IvyPatternRepositoryLayout>>()
+    fun ivyPatternLayout(action: Action<IvyPatternRepositoryLayout>) {
+        ivyPatternLayout.set(action)
+    }
+
```
Comment (marcphilipp) [inline src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishExtension.kt]: Why were these added again?

**Comment by maczikasz** [comment]
It seems I have started the branch from the wrong commit, will fix ASAP

File: src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt [hunk 3]
```diff
@@ -317,3 +317,8 @@ class NexusPublishPlugin : Plugin<Project> {
         }
     }
 }
+
+inline fun <reified T : Any> Project.theExtension(): T =
```
Comment (maczikasz) [inline src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt:321]: This issue in gradle https://github.com/gradle/gradle/issues/25206 causes a deprecation warning when using the `the` extension method to fetch extensions. This would cause tests to fail when running on gradle 8.2-milestoneX

Removing usage of the `the` method fixes this

File: src/main/kotlin/io/github/gradlenexus/publishplugin/NexusRepository.kt [hunk 4]
```diff
@@ -37,7 +37,7 @@ open class NexusRepository @Inject constructor(@Input val name: String, project:
     val snapshotRepositoryUrl = project.objects.property<URI>()
 
     @Input
-    val publicationType: Property<NexusPublishExtension.PublicationType> = project.objects.property<NexusPublishExtension.PublicationType>().value(NexusPublishExtension.PublicationType.MAVEN)
+    val publicationType: Property<NexusPublishExtension.PublicationType> = project.objects.property<NexusPublishExtension.PublicationType>().convention(NexusPublishExtension.PublicationType.MAVEN)
 
     @Internal
     val username = project.objects.property<String>().apply {
```
Comment (marcphilipp) [inline src/main/kotlin/io/github/gradlenexus/publishplugin/NexusRepository.kt:43]: I think we should use `convention` for the other properties such as this one as well. Unfortunately, `ProviderFactory.gradleProperty` was not added until Gradle 6.2.

File: src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt (hunk 3 above)
Comment (marcphilipp) [reply src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt:321]: Will this be fixed in Gradle 8.2 and we can remove the workaround?

**Review by marcphilipp — APPROVED** [review]

File: src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt (hunk 3 above)
Comment (maczikasz) [reply src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt:321]: That is not known currently, if yes we will be able to remove, otherwise it's easier to not rely on that extension
